qiskit/primitives/estimator_result.py

```diff
@@ -0,0 +1,40 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator result class
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+import numpy as np
+
+
+@dataclass(frozen=True)
+class EstimatorResult:
+    """
+    Result of ExpectationValue
+    #TODO doc
+    """
+
+    values: "np.ndarray[Any, np.dtype[np.float64]]"
+    variances: "np.ndarray[Any, np.dtype[np.float64]]"
```

Comment:
How about making `variances` optional, i.e., part of `metadata` or just remove?
There are some different options of error bound of expectation values. It might be convenient to allow developers to use their preferred error bound in the metadata or extend the result class as they wish.

Aer's instruction returns variance according to [api ref](https://qiskit.org/documentation/stubs/qiskit.providers.aer.library.SaveExpectationValueVariance.html)
Mitigator class and M3 return upper bound of standard deviation. [1. mitigator](https://qiskit.org/documentation/stubs/qiskit.result.LocalReadoutMitigator.expectation_value.html#qiskit.result.LocalReadoutMitigator.expectation_value), [2. M3](https://qiskit.org/documentation/partners/mthree/error.html)

Reply:
Sure, in interest of being minimal to start lets drop variance for first version

Reply:
I'm sure we will want to stadnardize at the ABC at least some common representations of errors, since the idea is to provide a baseline to alllow applications to be agnostic to the particular provider

Reply:
Thank you for the comment. We will remove `variance` as the first version. I agree that we need to standardize the representation of errors.